test/test-lex.cpp

```diff
@@ -196,13 +327,6 @@ TEST(test_lex, lex_invalid_big_int_number) {
                                 error_big_int_literal_contains_decimal_point,
                                 where, offsets_matcher(input, 0, 6))));
       });
-  check_tokens_with_errors(
-      u8"0123n", {token_type::number},
-      [](padded_string_view input, const auto& errors) {
-        EXPECT_THAT(errors, ElementsAre(ERROR_TYPE_FIELD(
-                                error_big_int_literal_contains_leading_zero,
```

Comment:
I think error_big_int_literal_contains_leading_zero is likely more helpful than error_unexpected_characters_in_octal_number. Can we easily make the lexer continue to report error_big_int_literal_contains_leading_zero?

Reply:
We'll see